v2/cmd/controller/app/setup.go

```diff
@@ -253,6 +254,37 @@ func SetupControllerManager(ctx context.Context, setupLog logr.Logger, flgs Flag
 	return mgr
 }
 
+func getMetricsOpts(flags Flags) server.Options {
+	var metricsOptions server.Options
+
+	if !flags.SecureMetrics {
```

Comment:
This code doesn't seem right, because it doesn't return here, it falls through and then overwrites the `metricsOptions` below?

I think you want something like:
```
var metricsOptions
if secure {
    // secure
} else {
    // insecure
}

if profiling {
    // profiling
}

return
```

Reply:
Or if you want to only enable profiling if secure, put that bit into the secure bit, and/or (maybe better) add a check that secure is not off with profiling on and if so return an error and crash the pod.

Reply:
Ah good catch. I missed return there. I'll update